examples/leds/leds_main.c

```diff
@@ -170,16 +197,25 @@ static int led_daemon(int argc, char *argv[])
           goto errout_with_fd;
         }
 
-      usleep(500*1000L);
+      usleep(500 * 1000L);
+    }
+
+  /* treats signal termination of the task */
+
+  if (g_led_daemon_started == false)
+    {
+      /* task terminated by a SIGTERM */
+
+      exit(EXIT_SUCCESS);
     }
 
-errout_with_fd:
-  close(fd);
+  errout_with_fd:
+    close(fd);
 
-errout:
-  g_led_daemon_started = false;
+  errout:
+    g_led_daemon_started = false;
```

Comment:
why change from line 176-182

Reply:
Actually there is no change except for indentation. 
The change is due to tools/checkpatch.sh message.

Reply:
But the old code looks good, what warning do you see?

Reply:
These lines:
  errout:
    g_led_daemon_started = false;

now are redundant, you can remove it

Reply:
In fact there were two odd error messages from checkpatch.sh.
One that related to the indentation of the very last "return" on lines 176-182.
The other saying that the path in line 2 (original source code) had to be related to "/" as it was not a nuttx path (or something like that). The point is that this error was related to a commentary block... 
In order to fix I had to change the commentaries to the APACHE message... even with a path, but after that the error was gone.

Reply:
> These lines:
> errout:
> g_led_daemon_started = false;
> 
> now are redundant, you can remove it

Not sure because there is a "goto errout;" in the code. The lines of code before that are executed when the while() loop exits due to change on g_led_daemon_started value change, in which case the function exits. Thus the errout: code will only be executed when an error occurs.

Reply:
right, you could just see the "errout:", although this redundant code is not an issue.

Reply:
Acctually it is related to exiting successfully after a normal termination with SIGTERM, or exiting with error code (goto errout: and other labels).

Reply:
I see! You are right! It needs to kept in case of failure during the start process!

Reply:
When exiting with SIGTERM it will be redundant, because g_led_daemon_started is already false (assuming you remove the "if (g_led_daemon_started == false)" block that I suggested.

Reply:
OK. I'll remove the "if (g_led ...)" 
I put it there actually  to make it explicit for others when reading the code, thus indeed redundant.
Most probably the compiler would optimize it anyway. I'll leave just the commentary for clearness.